proxy/http/HttpTransact.h

```diff
@@ -905,8 +905,9 @@ class HttpTransact
     int next_hop_scheme; // out
     int orig_scheme;     // pre-mapped scheme
     int method;
-    int cause_of_death_errno; // in
-    HostDBInfo host_db_info;  // in
+    int cause_of_death_errno;                         // in
+    Ptr<RefCountCacheItem<HostDBInfo>> *hostdb_entry; // Pointer to the entry we are referencing in hostdb-- to keep our ref
```

Comment:
``hostdb_entry`` doesn't need to be allocated, it should be ``Ptr<RefCountCacheItem<HostDBInfo>>``. Consider making a typedef for ``RefCountCacheItem<HostDBInfo>`` to reduce the verbosity.